x-pack/plugins/uptime/server/lib/adapters/framework/adapter_types.ts

```diff
@@ -9,14 +9,14 @@ import {
   IRouter,
   SavedObjectsClientContract,
   ISavedObjectsRepository,
-  ILegacyScopedClusterClient,
+  ElasticsearchClient,
 } from 'src/core/server';
 import { UMKibanaRoute } from '../../../rest_api';
 import { PluginSetupContract } from '../../../../../features/server';
 import { DynamicSettings } from '../../../../common/runtime_types';
 import { MlPluginSetup as MlSetup } from '../../../../../ml/server';
 
-export type ESAPICaller = ILegacyScopedClusterClient['callAsCurrentUser'];
+export type ESAPICaller = ElasticsearchClient;
```

Comment:
We likewise have a generic type called `UMElasticsearchQueryFn` which abstracts the need for the `request` functions to specify dependencies that are common to all functions of that type. Changing this type will update the expected type for each request object.